### PR TITLE
feat: add best_model_metrics to ModelCheckpoint callback

### LINKED ISSUE #19007: Add `best_model_metrics` to `ModelCheckpoint` callback
### Description & Motivation

There doesn't seem to be a straightforward way to get the training and validation metrics of the best model as monitored by ModelCheckpoint. Adding `best_model_metrics` attribute to `ModelCheckpoint` can be done easily in a way similar to how `best_model_score` works now.

### Pitch

This is useful for hyperparameter optimization if you want to optimize based on a validation metric or use other custom callbacks to summarize a batch experiment. I am willing to implement this feature.

### Alternatives

1. Running `trainer.test` on the training and validation dataset again after a `trainer.fit` call.
2. Getting records saved by loggers by checking the epoch number in the filename of a checkpoint file.

These two options create unnecessary overheads for a simple feature like this.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Iruos8805**
## What does this PR do?

This PR introduces a new `best_model_metrics` attribute to the ModelCheckpoint callback, enabling users to directly access all logged training and validation metrics associated with the best model, chosen based on the monitored metric.

Previously, only the `best_model_score` and `best_model_path` were available. With this addition, users can easily retrieve the corresponding metrics like val_loss, train_loss, val_metric etc. of the best checkpoint without reloading the model or rerunning validation.

This makes it easier to analyze results and use the best model’s metrics for reporting or hyperparameter tuning.

Fixes #19007 

<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21355.org.readthedocs.build/en/21355/

### CODE REVIEW

**Comment by Iruos8805** [comment]
@justusschock 

Thanks for the feedback. I’ll make the necessary changes.

**Comment by Iruos8805** [comment]
@justusschock 
I have applied the suggested changes and updated the tests and changelog.

**Comment by Iruos8805** [comment]
@justusschock 
Please let me know if the new changes look fine.
